### PR TITLE
Fix RSS title command-line argument injection (#1648)

### LINKED ISSUE #1648: Security: RSS feed title injects arbitrary arguments into desktop process on Windows
## Summary

A security review of Metalama''s network-facing code found that RSS feed title/text flows unescaped into a child-process command-line argument string on Windows. Split out from #1646 (Finding 2, MEDIUM).

**Threat model:** Metalama processes only trusted compilation inputs, so the relevant risk is a **hijacked/MITM''d backend server** (`metalama.net`), not malicious source code.

## Finding — RSS title/text → command-line argument injection into the desktop process (MEDIUM)

`WindowsUserInterfaceService.ShowToastNotification` builds the child-process argument string by raw interpolation with no escaping (`WindowsUserInterfaceService.cs:36-49`):

```csharp
arguments += $" --title \"{notification.Title}\"";   // Title is the raw RSS <title>
arguments += $" --uri \"{notification.Uri}\"";
```

The RSS `<title>` is not normalized (only the URI passes through `UriBuilder`), so a title containing a `"` breaks out of the quoted argument and injects arbitrary additional arguments into `Metalama.Backstage.Desktop.Windows.exe`, which runs with `UseShellExecute = true` (`BackstageTool.cs:22`).

`ShellExecute` means no `cmd.exe` interpretation (so not arbitrary shell-command injection), but the attacker fully controls the desktop tool''s `argv` — including injecting their own `--uri "<dangerous-scheme>"`. This means the scheme check from the RSS-scheme finding alone is insufficient; both layers need hardening.

## Fix

- Escape embedded quotes, or (preferred) move to `ProcessStartInfo.ArgumentList` end-to-end.
- Sanitize `Title`/`Text` before they reach the argument string.
- Add regression tests in `RssClientTests` (quote-injection title).

Related: split out from #1646 alongside the RSS protocol-activation finding (HIGH).

— Claude for Gael

### PR DESCRIPTION
**Pull request by PostSharpAgent**
## Summary

Fixes #1648 ΓÇö **Security (MEDIUM): RSS feed title injects arbitrary arguments into the desktop process on Windows.**

`WindowsUserInterfaceService.ShowToastNotification` built the child-process (`Metalama.Backstage.Desktop.Windows.exe`) command line by interpolating the untrusted toast `Title`/`Text`/`Uri` into a string with no escaping. A `Title` from a hijacked/MITM'd RSS feed containing a `"` could break out of the quoted `--title` argument and inject arbitrary `argv` (e.g. an attacker-controlled `--uri`).

## Fix

Pass arguments as a vector end-to-end instead of a concatenated string:
- `IBackstageToolsExecutor.Start( BackstageTool, params string[] arguments )`.
- `BackstageToolsExecutor` quotes each argument with the canonical Windows algorithm (round-trips through `CommandLineToArgvW`). `ProcessStartInfo.ArgumentList` is not usable because the assembly also targets `netframework4.7.2`/`netstandard2.0`.
- `WindowsUserInterfaceService` and the `web` server call site pass separate tokens.

## Tests

`WindowsUserInterfaceServiceTests`: title injection (incl. backslash/quote edge cases), text injection, benign baseline. Verified the test fails before the fix (injected a second `--uri`) and passes after. 13 affected tests green on `net8.0` and `netframework4.7.2`.

## Checklist

- [x] Phase 1 ΓÇö Understand the issue
- [x] Phase 2 ΓÇö Failing regression test reproducing the injection
- [x] Phase 3 ΓÇö Fix
- [x] Phase 4 ΓÇö Full build & test (zero warnings)
- [x] Phase 5 ΓÇö Finalize (marked ready, review requested)

ΓÇö Claude for Gael

### CODE REVIEW

File: Metalama.Backstage/src/tests/Metalama.Backstage.Tests/UserInterface/WindowsUserInterfaceServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    public static IEnumerable<object[]> MaliciousTitles()
+    {
+        // Break out of the quoted argument and inject a second --uri.
+        yield return ["Pwned\" --uri \"" + _injectedUri];
+
+        // A trailing backslash before the closing quote must not let the quote be escaped away.
+        yield return ["Pwned\\\" --uri \"" + _injectedUri];
+
+        // Multiple backslashes followed by a quote.
+        yield return ["Pwned\\\\\" --uri \"" + _injectedUri];
```
[the remark this answers is not in the document]
Comment (gfraiteur) [reply Metalama.Backstage/src/tests/Metalama.Backstage.Tests/UserInterface/WindowsUserInterfaceServiceTests.cs:45]: Irrelevant.

**Review by gfraiteur — CHANGES_REQUESTED** [review]
Implement copilot's feedback except where dismissed.

File: Metalama.Backstage/src/Metalama.Backstage/Tools/BackstageToolsExecutor.cs [hunk 2]
```diff
@@ -54,22 +56,108 @@ public IProcess Start( BackstageTool tool, string arguments )
...
+    private static void AppendArgument( StringBuilder builder, string argument )
+    {
+        if ( builder.Length != 0 )
+        {
+            builder.Append( ' ' );
+        }
+
+        // An argument with no whitespace or quote can be appended verbatim.
+        if ( argument.Length != 0 && argument.IndexOfAny( new[] { ' ', '\t', '\n', '\v', '"' } ) < 0 )
+        {
```
[the remark this answers is not in the document]
Comment (PostSharpAgent) [reply Metalama.Backstage/src/Metalama.Backstage/Tools/BackstageToolsExecutor.cs:111]: Done in cb00303787 — hoisted the character set into a `static readonly char[] _charactersRequiringQuoting` field and reuse it in `AppendArgument.IndexOfAny`, removing the per-call allocation.

— Claude for Gael

**Review by gfraiteur — APPROVED** [review]
